jdaviz/configs/cubeviz/plugins/tools.py

```diff
@@ -127,12 +128,30 @@ def on_mouse_move(self, data):
         x = int(np.round(data['domain']['x']))
         y = int(np.round(data['domain']['y']))
 
-        # Use first visible layer for now
-        cube_data = [layer.layer for layer in self.viewer.layers if layer.state.visible
-                     and len(layer.layer.shape) == 3]
-        if len(cube_data) == 0:
-            return
-        cube_data = cube_data[0]
+        # Use the selected layer from coords_info as long as it's 3D
+        coords_info_dataset = self.viewer.session.application._tools['g-coords-info'].dataset.selected
+        if coords_info_dataset == 'auto':
+            cube_data = self.viewer.active_image_layer.layer
+        elif coords_info_dataset == 'none':
+            cube_data = self.viewer.layers[0].layer
+        else:
+            for layer in self.viewer.layers:
+                if layer.layer.label == coords_info_dataset and layer.visible:
+                    if isinstance(layer, BqplotImageSubsetLayerArtist):
+                        # cannot expose info for spatial subset layers
+                        continue
+                    cube_data = layer.layer
+                    break
```

Comment:
can this rely on `dataset.selected_obj` (which will then get caching for free)?  Or do you need some layer info that isn't accessible there?

Reply:
That also then might let you mouseover the uncertainty viewer but access the mouseover from the flux cube (if selected manually in the mouseover cycler) which would be super cool!

Reply:
I was just copying the logic from `coords_info`, using `selected_obj` seems to work with an extra check for glue Data vs Spectrum1D. I do worry that it will be confusing for users rather than a good feature that if they select the hover option in a viewer, they might be seeing the data from another viewer if it's selected at the top level.

Reply:
hmmm true, I think most people don't even use the cycler though, so probably won't run into this except for advanced use-cases.  Might be worth some user-testing or just see if anyone complains about confusion.